bulbs/recirc/mixins.py

```diff
@@ -0,0 +1,91 @@
+from django.conf import settings
+from django.db import models
+
+from elasticsearch import Elasticsearch
+from json_field import JSONField
+
+from bulbs.content.custom_search import custom_search_model
+from bulbs.content.models import Content
+
+
+es = Elasticsearch(settings.ES_CONNECTIONS["default"]["hosts"])
+
+
+class BaseQueryMixin(models.Model):
+
+    query = JSONField(default={}, blank=True)
+
+    class Meta:
+        abstract = True
+
+    def clean(self):
+        super(BaseQueryMixin, self).clean()
+        self.clean_query()
+
+    def clean_query(self):
+        """
+        Removes any `None` value from an elasticsearch query.
+        """
+        if self.query:
+            for key, value in self.query.items():
+                if isinstance(value, list) and None in value:
+                    self.query[key] = [v for v in value if v is not None]
+
+    def save(self, *args, **kwargs):
+        self.clean()
+        super(BaseQueryMixin, self).save()
+
+    def get_recirc_content(self, published=True):
+        """gets the first 3 content objects in the `included_ids`
+        """
+
+        # NOTE: set included_ids to just be the first 3 ids,
+        # otherwise search will return last 3 items
+        if "query" in self.query:
```

Comment:
@mparent61 @benghaziboy updated this part to set explicitly set `included_ids` to the first 3 ids in the query, since `custom_model_search` was only returning the last 3 objects